lib/vikings.ts

```diff
@@ -0,0 +1,51 @@
+import { Viking } from "../types/viking";
+type vikingDataJSON = {
+  fullName: string;
+  village: string;
+  age: number;
+  valkyrie: string;
+  hasSon: boolean;
+};
+type vikingDataYaml = {
+  name: string;
+  weapon: "sword" | "spear" | "axe" | "hammer";
+  name_of_father: string;
+  has_home_in: string;
+  years_old: number;
+  number_of_children: number;
+};
+
+export const handleVikingList = (
+  vikingsJSON: Array<vikingDataJSON>,
+  vikingsYaml: Array<vikingDataYaml>
+) => {
```

Comment:
You could write return type here (: Viking[]) and then you could remove all those Array<Viking> from inside the function and it would work. You don't have to use type assertion everywhere for every variable, Typescript is clever enough to infer that :)